autoload/neomake/makers/ft/sh.vim

```diff
@@ -4,14 +4,35 @@ function! neomake#makers#ft#sh#EnabledMakers() abort
     return ['sh', 'shellcheck']
 endfunction
 
-" IDEA: could be detected once from "shellcheck --help" (since older versions
-" support zsh, and newer might do so again).
-let s:shellcheck_supported = ['sh', 'bash', 'dash', 'ksh']
+function! s:shellcheck_determine_supported() abort
+    let s:shellcheck_supported = ['sh', 'bash', 'dash', 'ksh']
+
+    " zsh support was available from the first release of shellcheck and
+    " removed in version 0.3.6. Before shellcheck version 0.3.1 there was no
+    " way to get the version information.
+    let l:version_line = matchstr(split(system('shellcheck --version'), '\n'),
+                \ '^version:')
+    if v:shell_error != 0
+        let s:shellcheck_supported += ['zsh']
```

Comment:
Unfortunately that will not work for all versions. I just verified the exit codes for versions 0.1.0 & 0.2.0. On version 0.1.0 there is no failure for the exit code, rather `--version` is considered a missing file with an exit code of 0.

```
Downloads/shellcheck-0.1.0$ ./shellcheck --version
No such file: --version
Downloads/shellcheck-0.1.0$ echo $?
0
```

While version 0.2.0 has an exit status of 1:
```
Downloads/shellcheck-0.2.0$ ./shellcheck --version
No such file: --version
Downloads/shellcheck-0.2.0$ echo $?
1
```

My code does not correctly support shellcheck version 0.1.0, so I'll update the code for that case.